doc/nrf/device_guides/working_with_nrf/nrf53/qspi_xip_guide.rst

```diff
@@ -0,0 +1,324 @@
+
+* The QSPI memory starting with the 0x10000000 internal memory address.
+* The offset of an external application part image within the QSPI flash.
+  The external application code partition is mapped by the ``mcuboot_primary_2`` PM partition.
+* The image header size of MCUboot (0x200).
+
+See the following example of the calculation:
+
+.. code-block:: console
+
+    #include <zephyr/linker/sections.h>
+    #include <zephyr/devicetree.h>
+    #include <zephyr/linker/linker-defs.h>
+    #include <zephyr/linker/linker-tool.h>
+
+    MEMORY
+    {
+        /* This maps in mcuboot_primary_2 partition defined in pm_static.yaml
+        * components for ORIGIN calculation:
+        *  - 0x10000000: offset of QSPI external memory in SoC memory mapping.
+        *  - 0x120000: mcuboot_primary_2 offset in QSPI external memory
+        *  - 0x200: image header size.
+        * The size of this region is size of mcuboot_primary_2 reduced by the
+        * image header size.
+        */
+        EXTFLASH (wx) : ORIGIN = 0x10120200, LENGTH = 0x3FE00
+    }
+
+    #include <zephyr/arch/arm/cortex_m/scripts/linker.ld>
+
+Code relocation setup
+=====================
+
+Relocating code to QSPI XIP is a part of the project's :file:`CMakeLists.txt` file.
+You can set up the relocation on a file or library basis using the ``zephyr_code_relocate()`` function.
+For example, to relocate a file in the application, you can use the following configuration:
+
+.. code-block:: console
+
+   zephyr_code_relocate(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth.c LOCATION EXTFLASH_TEXT NOCOPY)
+   zephyr_code_relocate(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth.c LOCATION RAM_DATA)
+
+Where the first line relocates the XIP code (.text) and the second line relocates the data initialization content section (.data).
+
+Similarly, it is possible to relocate certain libraries, for example:
+
+.. code-block:: console
+
+   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION EXTFLASH_TEXT NOCOPY)
+   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION RAM_DATA)
+
+Flashing the project
+====================
+
+For the nRF5340 DK and other boards equipped with flash working in the QSPI mode, use the `west flash` command.
+For other cases, flashing needs to be done manually.
+
+Flashing to external flash in SPI/DSPI mode
+-------------------------------------------
+
+When flashing an application using ``west``, the ``nrfjprog`` runner is invoked.
+The runner uses the system default configuration that configures the application in the QSPI mode when flashing the external flash.
+
+Troubleshooting
+===============
+
+Refer to the following sections for the most common issues.
+
+Module does not appear to start
+-------------------------------
+
+A common issue when using QSPI XIP in an application is that the module does not appear to start or continuously ends in a fault before the application can run.
```

Comment:
Perhaps lines 304-307 can be slightly modified to the following?

> When using QSPI XIP, a frequent issue is the module not starting or crashing before the app runs.
> This often results from a mismatch in ``init`` priorities between the code on QSPI flash and the QSPI flash device.

Reply:
Sounds good!